File: src/test/java/org/oewntk/yaml/in/TestsYamlModelWords.java
/*
 * Copyright (c) 2021. Bernard Bou.
 */

package org.oewntk.yaml.in;

import org.junit.BeforeClass;
import org.junit.Test;
import org.oewntk.model.Key;
import org.oewntk.model.LibTestModel;

import java.util.Set;

public class TestsYamlModelWords
{
	private static final boolean peekTestWords = false;

	private static final Set<String> testWords = Set.of("baroque", "Baroque", "allied", "Allied", "earth", "Earth", "bass", "row", "critical");

	@BeforeClass
	public static void init()
	{
		TestsYamlCommon.init();
	}

	@Test
	public void testScanLexesForTestWords()
	{
		LibTestModel.testScanLexesForTestWords(TestsYamlCommon.model, Key.OEWN::of, LibTestModel::makeIndexMap, testWords, peekTestWords, TestsYamlCommon.ps);
	}

	@Test
	public void testScanLexesForTestWordsSorted()
	{
		LibTestModel.testScanLexesForTestWords(TestsYamlCommon.model,  Key.OEWN::of, LibTestModel::makeSortedIndexMap, testWords, peekTestWords, TestsYamlCommon.ps);
	}

	@Test
	public void testBass()
	{
		LibTestModel.testWord("bass", TestsYamlCommon.model, TestsYamlCommon.ps);
	}

	@Test
	public void testRow()
	{
		LibTestModel.testWord("row", TestsYamlCommon.model, TestsYamlCommon.ps);
	}

	@Test
	public void testBaroque()
	{
		LibTestModel.testWords(TestsYamlCommon.model, TestsYamlCommon.ps, "baroque", "Baroque");
	}

	@Test
	public void testEarth()
	{
		LibTestModel.testWords(TestsYamlCommon.model, TestsYamlCommon.ps, "earth", "Earth");
	}

	@Test
	public void testCritical()
	{
		LibTestModel.testWord("critical", 'a', TestsYamlCommon.model, TestsYamlCommon.ps);
	}

	@Test
	public void testHollywood()
	{
		LibTestModel.testWord("Hollywood", 'a', TestsYamlCommon.model, TestsYamlCommon.ps);
	}

	@Test
	public void testVictorian()
	{
		LibTestModel.testWord("Victorian", 'a', TestsYamlCommon.model, TestsYamlCommon.ps);
	}

	@Test
	public void testAllied()
	{
		LibTestModel.testWord("allied", 'a', TestsYamlCommon.model, TestsYamlCommon.ps);
	}

	@Test
	public void testAlliedUpper()
	{
		LibTestModel.testWord("Allied", 'a', TestsYamlCommon.model, TestsYamlCommon.ps);
	}

	@Test
	public void testAbsent()
	{
		LibTestModel.testWord("absent", 'a', TestsYamlCommon.model, TestsYamlCommon.ps);
	}

	@Test
	public void testApocryphal()
	{
		LibTestModel.testWord("apocryphal", 'a', TestsYamlCommon.model, TestsYamlCommon.ps);
	}

	@Test
	public void testUsed()
	{
		LibTestModel.testWord("used", 'a', TestsYamlCommon.model, TestsYamlCommon.ps);
	}
}

File: src/test/java/org/oewntk/yaml/in/TestsYamlModelWords.kt
/*
 * Copyright (c) 2021-2024. Bernard Bou.
 */
package org.oewntk.yaml.`in`

import org.junit.BeforeClass
import org.junit.Test
import org.oewntk.model.Key
import org.oewntk.model.Lemma
import org.oewntk.model.Lex
import org.oewntk.model.LibTestModel.makeIndexMap
import org.oewntk.model.LibTestModel.makeSortedIndexMap
import org.oewntk.model.LibTestModel.testScanLexesForTestWords
import org.oewntk.model.LibTestModel.testWord
import org.oewntk.model.LibTestModel.testWords
import org.oewntk.model.PartOfSpeech
import org.oewntk.yaml.`in`.LibTestsYamlCommon.model
import org.oewntk.yaml.`in`.LibTestsYamlCommon.ps

class TestsYamlModelWords {

    @Test
    fun testScanLexesForTestWords() {
        testScanLexesForTestWords(model, { lex: Lex -> Key.UsingPronunciation.of(lex) }, { seq: Sequence<Key> -> makeIndexMap(seq) }, testWords, PRINT_TEST_WORDS, ps)
    }

    @Test
    fun testScanLexesForTestWordsSorted() {
        testScanLexesForTestWords(model, { lex: Lex -> Key.UsingPronunciation.of(lex) }, { seq: Sequence<Key> -> makeSortedIndexMap(seq) }, testWords, PRINT_TEST_WORDS, ps)
    }

    @Test
    fun testBass() {
        testWord(Lemma("bass"), model, ps)
    }

    @Test
    fun testRow() {
        testWord(Lemma("row"), model, ps)
    }

    @Test
    fun testCentre() {
        testWord(Lemma("centre"), model, ps)
    }

    @Test
    fun testBaroque() {
        testWords(model, ps, arrayOf(Lemma("baroque"), Lemma("Baroque")))
    }

    @Test
    fun testEarth() {
        testWords(model, ps, arrayOf(Lemma("earth"), Lemma("Earth")))
    }

    @Test
    fun testCritical() {
        testWord(Lemma("critical"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testHollywood() {
        testWord(Lemma("Hollywood"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testVictorian() {
        testWord(Lemma("Victorian"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testAllied() {
        testWord(Lemma("allied"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testAlliedUpper() {
        testWord(Lemma("Allied"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testAbsent() {
        testWord(Lemma("absent"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testApocryphal() {
        testWord(Lemma("apocryphal"), PartOfSpeech.A, model, ps)
    }

    @Test
    fun testUsed() {
        testWord(Lemma("used"), PartOfSpeech.A, model, ps)
    }

    companion object {

        private const val PRINT_TEST_WORDS = false

        private val testWords = setOf(Lemma("baroque"), Lemma("Baroque"), Lemma("allied"), Lemma("Allied"), Lemma("earth"), Lemma("Earth"), Lemma("bass"), Lemma("row"), Lemma("critical"), Lemma("centre"))

        @JvmStatic
        @BeforeClass
        fun init() {
            model
        }
    }
}
